dev/archery/archery/benchmark/core.py

```diff
@@ -27,11 +27,14 @@ def median(values):
 
 
 class Benchmark:
-    def __init__(self, name, unit, less_is_better, values, counters=None):
```

Comment:
Manually tested this change with the following command.

`$ archery benchmark run --suite-filter=arrow-compute-vector-selection-benchmark --benchmark-filter=TakeStringRandomIndicesWithNulls --repetitions=3`

Output snippet:

```
{
    "suites": [
        {
            "name": "arrow-compute-vector-selection-benchmark",
            "benchmarks": [
                {
                    "name": "TakeStringRandomIndicesWithNulls/262144/0",
                    "unit": "items_per_second",
                    "less_is_better": false,
                    "values": [
                        18536824.77866367,
                        19191618.243985265,
                        19796322.323237784
                    ],
                    "time_unit": "ns",
                    "times": [
                        14596346.90588616,
                        14821237.647319557,
                        16059698.500342807
                    ]
                },
                
                ...
                
            ]
        }
    ]
}                
```

Reply:
Because this shape change is additive, `archery benchmark diff` will still work.

The downside, is that I would also like to eventually expose both `items_per_second ` and `bytes_per_second` at some point, and the current shape doesn't lend itself well to multiple benchmark units.

I suppose an alternative would be to not merge this commit at all, and instead add a command line option to just exposes the unadulterated google benchmark output which has all I need. For example:

```
 {'cpu_time': 12718379.310344812,
  'items_per_second': 20611431.18972546,
  'iterations': 58,
  'name': 'TakeStringRandomIndicesWithNulls/262144/0',
  'null_percent': 0.0,
  'real_time': 13292835.792526603,
  'repetition_index': 0,
  'repetitions': 0,
  'run_name': 'TakeStringRandomIndicesWithNulls/262144/0',
  'run_type': 'iteration',
  'size': 262144.0,
  'threads': 1,
  'time_unit': 'ns'},
```

Reply:
Interesting. Did you check it works properly with a benchmark that only outputs runtimes? One example is `arrow-memory-pool-benchmark`.

Reply:
It will work fine, but it will be redundant.

That said, I just tested the suite you mentioned, and it looks like I've found an unrelated bug.

I would have expected `times` to be equal to `values` here because the benchmark name ends in `real_time` which is supposed to be a hint to prefer `real_time` over `cpu_time`, but the check is using `/realtime"`, not `/real_time"`.

I'lll create a different issue and pull request for that since it's unrelated to this one.

```
    @property
    def is_realtime(self):
        """ Indicate if the preferred value is realtime instead of cputime. """
        return self.name.find("/realtime") != -1
```

```
                {
                    "name": "AllocateDeallocate<Jemalloc>/size:1048576/real_time",
                    "unit": "ns",
                    "less_is_better": true,
                    "values": [
                        1508.9095897178113,
                        1511.4170658339356,
                        1524.9203870819892
                    ],
                    "time_unit": "ns",
                    "times": [
                        1513.748357015267,
                        1515.7354204676608,
                        1534.3125060644313
                    ]
                },
```

Reply:
New arrow issues created:

[Developer][Archery] Fix prefer real time check
https://issues.apache.org/jira/browse/ARROW-11746

Reply:
PS. If I fix the prefer real time check, I see the redundant results I would expect for benchmarks ending in `/real_time`.

```
                {
                    "name": "AllocateDeallocate<Jemalloc>/size:1048576/real_time",
                    "unit": "ns",
                    "less_is_better": true,
                    "values": [
                        1669.6396259875594,
                        1681.1759457288072,
                        1689.4427232189955
                    ],
                    "time_unit": "ns",
                    "times": [
                        1669.6396259875594,
                        1681.1759457288072,
                        1689.4427232189955
                    ]
                },
```

```
     @property
     def is_realtime(self):
         """ Indicate if the preferred value is realtime instead of cputime. """
-        return self.name.find("/realtime") != -1
+        return self.name.find("/real_time") != -1
```

Reply:
Well, I don't think the redundancy hurts either, so it'd be fine with me.